bitbots_vision/src/bitbots_vision/vision_modules/yolo_handler.py

```diff
@@ -485,87 +487,123 @@ def set_image(self, image):
         """
         self._yolo.set_image(image)
 
+    @abc.abstractmethod
     def get_candidates(self):
         """
-        :return: all found ball candidates
+        :return: all found candidates
         """
-        return self._yolo.get_candidates("ball")
+        raise NotImplementedError
 
     def compute(self):
         """
         Runs the yolo network
         """
         self._yolo.predict()
 
-class YoloGoalpostDetector(CandidateFinder):
+class YoloBallDetector(YoloDetector):
     """
-    A goalpost detector using the yolo neural network.
+    A ball detector using the yolo neural network.
     This layer connects a single YOLO network with multiple candidate finders for the different classes,
```

Comment:
Is layer the right word here?

Reply:
Its an abstraction layer in the software architecture. Not a layer in the context of YOLO/ML